python/pyarrow/_parquet.pyx

```diff
@@ -342,13 +343,13 @@ cdef logical_type_name_from_enum(ParquetLogicalType type_):
 cdef class ParquetReader:
     cdef:
         object source
-        MemoryPool* allocator
+        CMemoryPool* allocator
         unique_ptr[FileReader] reader
         column_idx_map
         FileMetaData _metadata
 
-    def __cinit__(self):
-        self.allocator = default_memory_pool()
+    def __cinit__(self, MemoryPool memory_pool=None):
```

Comment:
I had thought about granularity of memory pools in PyArrow. While having a global memory pool that you can change would be convenient, it could also be useful to create "subpools" so that you can more easily track memory use within a particular domain of work. 

There is also the matter of keeping the `arrow::MemoryPool` instance alive (`PoolBuffer` does not share ownership of the memory pool)